js_modules/dagster-ui/packages/ui-core/src/assets/RecentUpdatesTimeline.tsx

```diff
@@ -0,0 +1,209 @@
+import {useRecentAssetEvents} from './useRecentAssetEvents';
+import {Timestamp} from '../app/time/Timestamp';
+import {AssetRunLink} from '../asset-graph/AssetRunLinking';
+import {RunStatusWithStats} from '../runs/RunStatusDots';
+import {titleForRun} from '../runs/RunUtils';
+import {useFormatDateTime} from '../ui/useFormatDateTime';
+
+const MIN_TICK_WIDTH = 5;
+
+export const RecentUpdatesTimeline = ({
+  assetKey,
+  materializations,
+}: {
+  assetKey: AssetKey;
+  materializations: ReturnType<typeof useRecentAssetEvents>['materializations'];
+}) => {
+  const {containerProps, viewport} = useViewport();
+  const widthAvailablePerTick = viewport.width / 100;
+
+  const tickWidth = Math.max(widthAvailablePerTick, MIN_TICK_WIDTH);
+
+  const buckets = Math.floor(viewport.width / tickWidth);
+
+  const endTimestamp = parseInt(materializations[0]!.timestamp);
+  const startTimestamp = parseInt(materializations[materializations.length - 1]!.timestamp);
+  const range = endTimestamp - startTimestamp;
+
+  const bucketedMaterializations = useMemo(() => {
+    if (!viewport.width) {
+      return [];
+    }
+    const firstPassBucketsArray: Array<{
+      number: number;
+      materializations: typeof materializations;
+    }> = new Array(buckets);
+
+    materializations.forEach((materialization) => {
+      const bucketNumber = Math.floor(
+        ((parseInt(materialization.timestamp) - startTimestamp) / range) * buckets,
+      );
+      firstPassBucketsArray[bucketNumber] = firstPassBucketsArray[bucketNumber] || {
+        number: bucketNumber,
+        materializations: [] as typeof materializations,
+      };
+      firstPassBucketsArray[bucketNumber]!.materializations.push(materialization);
+    });
+
+    const secondPassBucketsArray: Array<{
+      start: number;
+      end: number;
+      materializations: typeof materializations;
+    }> = [];
+
+    firstPassBucketsArray.forEach((bucket) => {
+      const lastBucket = secondPassBucketsArray[secondPassBucketsArray.length - 1];
+      if (!lastBucket || lastBucket.end !== bucket.number - 1) {
+        secondPassBucketsArray.push({
+          start: bucket.number,
+          end: bucket.number,
+          materializations: bucket.materializations,
+        });
+      } else {
+        lastBucket.end = bucket.number;
+        lastBucket.materializations = [...lastBucket.materializations, ...bucket.materializations];
+      }
+    });
+    return secondPassBucketsArray;
+  }, [viewport.width, buckets, materializations, startTimestamp, range]);
+
+  const formatDateTime = useFormatDateTime();
+
+  return (
+    <Box flex={{direction: 'column', gap: 4}}>
+      <Box flex={{direction: 'row', justifyContent: 'space-between'}}>
+        <Subtitle2>Recent updates</Subtitle2>
+        <Caption color={Colors.textLighter()}>
+          {materializations.length === 100
+            ? 'Last 100 updates'
+            : `Showing all ${materializations.length} updates`}
```

Comment:
Possible to have zero or one materialization?

Reply:
Not 0, but yeah 1, I'll update this 👍🏾